maintenance/health/servicehealthcheck/healthchecker.go

```diff
@@ -41,31 +45,68 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	name := splitRoute[1]
-	hcInterface, isIn := checks.Load(name)
-	if !isIn {
-		h.writeError(w, errors.New("health check not registered"), http.StatusNotFound, name)
-		return
+	hcs := &checks
+	checkAll := name == nameAllHealthCheck
+	if !checkAll {
+		hcInterface, isIn := checks.Load(name)
+		if !isIn {
+			h.writeError(w, errors.New("health check not registered"), http.StatusNotFound, name)
+			return
+		}
+		hc := hcInterface.(HealthCheck)
```

Comment:
It's not necessary to check the type here. The value will not lose its type if you don't.